scripts/operations_tests/operations_tests.py

```diff
@@ -222,30 +239,43 @@ def load_post_operation_image_stack(filepath):
 
 
 def load_image_stack():
-    filename_group = FilenameGroup.from_file(config_manager.load_sample)
-    filename_group.find_all_files()
-    image_stack = loader.load(filename_group=filename_group)
-    return image_stack
+    try:
+        load_sample_path = config_manager.load_sample
+        filename_group = FilenameGroup.from_file(load_sample_path)
+        filename_group.find_all_files()
+        image_stack = loader.load(filename_group=filename_group)
+        return image_stack
+    except FileNotFoundError as err:
+        error_msg = f"File not found at path: {load_sample_path}"
+        logging.error(error_msg)
+        raise FileNotFoundError(error_msg) from err
+    except Exception as err:
+        error_msg = f"Error while loading image stack: {err}."
+        logging.error(error_msg)
+        raise Exception(error_msg) from err
 
 
 def compare_image_stacks(baseline_image_stack, new_image_stack, test_case):
-    if not (isinstance(baseline_image_stack, np.ndarray) and isinstance(new_image_stack, np.ndarray)):
-        test_case.status = "fail"
-        test_case.message = "new image stack is not an array"
-    elif baseline_image_stack.shape != new_image_stack.shape:
-        test_case.status = "fail"
-        test_case.message = "new image stack is different shape to the baseline"
-    elif baseline_image_stack.dtype != new_image_stack.dtype:
-        test_case.status = "fail"
-        test_case.message = "new image stack is different dtype to the baseline"
-    elif not np.array_equal(baseline_image_stack, new_image_stack):
-        test_case.status = "fail"
-        test_case.message = "arrays are not equal"
-        if args.gui:
-            gui_compare_image_stacks(baseline_image_stack, new_image_stack)
-    else:
+    try:
+        if not isinstance(baseline_image_stack, np.ndarray) or not isinstance(new_image_stack, np.ndarray):
+            raise ValueError("new image stack is not an array")
+
+        if baseline_image_stack.shape != new_image_stack.shape:
+            raise ValueError("new image stack is different shape to the baseline")
+
+        if baseline_image_stack.dtype != new_image_stack.dtype:
+            raise ValueError("new image stack is different dtype to the baseline")
+
+        if not np.array_equal(baseline_image_stack, new_image_stack):
+            raise ValueError("Image stacks are not identical, arrays are not equal")
+
         test_case.status = "pass"
-        test_case.message = "arrays are equal"
+        test_case.message = "Image stacks are identical."
+
+    except ValueError as e:
+        test_case.status = "fail"
+        test_case.message = str(e)
+        logging.error(f"Comparison failed: {e}")
```

Comment:
I think I prefer this as regular flow control, rather than using exceptions and catching them straight away.